src/when.js

```diff
@@ -27,7 +27,10 @@ class WhenMock {
     this.callMocks = []
 
     const mockReturnValue = (matchers, assertCall, once = false) => (val) => {
-      this.callMocks.push({ matchers, val, assertCall, once })
+      // Remove call mocks with equal matchers to enable dynamic replacement during a test
+      this.callMocks = this.callMocks
+        .filter((callMock) => once || !utils.equals(callMock.matchers, matchers))
+        .concat({ matchers, val, assertCall, once })
```

Comment:
The implementation of the `*Once` types is a nice addition to the module, but it's kind of incomplete.

Usually i use e.g. `mockReturnValueOnce` in combination with `mockReturnValue`. I setup a `jest.fn()` for the whole test suite which returns a defined value which fits for most of my test cases. But within single tests (usually the error cases) it needs to return something else. Within those situations i use `mockReturnValueOnce` to get a different return value once from the mock.

That doesn't work correct with the current implementation, since the `mockReturnValue` comes first within the `callMocks`. Example setup:
```
when(fn).calledWith('foo').mockReturnValue('1337')
when(fn).calledWith('foo').mockReturnValueOnce('bar')
when(fn).calledWith('foo').mockReturnValueOnce('cbs')
```
`fn('foo')` called three times will always return "1337". After the 2nd call the `mockReturnValueOnce` are all removed, since the logic pops one entry per call from the stack, which has correct matchers and a `once` property.

This additional step could be a solution, but i only roughly tested it
```
.sort(({ once }) => {
  // Sort entries without the flag to the end of the list, but keep the order of the others
  return !once ? 1 : 0
})
```

Reply:
To be honest, I didn't think about _that_ use case, because mostly I only used `mockReturnValue` OR `mockReturnValueOnce`. If I used both I used them in the other order as you mention it

But yes, your use case makes absolutely sense and your proposed additional step seems reasonable. I'll see what I can do based on your "roughly tested" implementation 👍